### ISSUE TITLE
 Fatal error (ignored): missing PATH-TO-LICENSE-FILE argument

### ISSUE THREAD

**Issue by GrahamHannington** [splunk/docker-splunk#67]
Here's an excerpt from the log from a Docker container that is based on the Splunk 7.2.0 Docker image:

````
TASK [splunk_common : Apply Splunk license] ************************************
Tuesday 27 November 2018  07:45:54 +0000 (0:00:00.104)       0:01:30.849 ******
fatal: [localhost]: FAILED! => {"changed": true, "cmd": ["/opt/splunk/bin/splunk", "add", "licenses", "-auth", "admin:changeme"], "delta": "0:00:03.012263", "end": "2018-11-27 07:45:57.518058", "msg": "non-zero return code", "rc": 4, "start": "2018-11-27 07:45:54.505795", "stderr": "", "stderr_lines": [], "stdout": "missing PATH-TO-LICENSE-FILE argument: ./splunk add license [PATH-TO-FILE] ", "stdout_lines": ["missing PATH-TO-LICENSE-FILE argument: ./splunk add license [PATH-TO-FILE] "]}
...ignoring
````

Is there a way to avoid or suppress that "fatal" error?

The Docker container starts okay, so... there's actually no real problem here. I just don't like that "fatal" error appearing in the log.

Here's why: I'm going to publish a Docker image that is based on this Splunk Docker image. In the documentation that accompanies my image, I have a troubleshooting section that asks users to send the output from `docker logs`. Users might notice that "fatal" error, ask about it, perhaps point that out as the potential problem. I'd prefer not to have to deal with that.

Possible alternatives to suppressing that error:

* Would you consider changing the error message severity level so that it's not reported as "fatal"? If an error occurs that you can ignore, and then continue processing, then, by any definition I'm aware of, _it's not fatal_.
* Is there a way to start a container with a Free license, rather than, as now, an Enterprise Trial license? I haven't looked into the contents of license files. Would it be possible to supply a license file that says "this is a Splunk Free license"? (Or is it the case that Splunk Free doesn't require/use a license file?)

**Comment by GrahamHannington** [comment]
In my previous comment, I wrote:

> I'm going to publish a Docker image that is based on this Splunk Docker image

Here it is:

https://hub.docker.com/r/fundisoftware/taw-splunk/

Actually, that image has been published for a while. I've just updated it to use Splunk 7.2.0. Previously, it was using Splunk 6.6.3. I updated it both to stay current, and because the Splunk 6.6.3 Enterprise Trial license was "pre-expired" on starting new containers. For details, see my related question in Splunk Answers, "[Why has the Splunk Enterprise Trial license already expired in a brand new Docker container?](https://answers.splunk.com/answers/670551/)".
